experimental/dds/tree2/src/simple-tree/proxies.ts

```diff
@@ -815,23 +815,66 @@ function createMapProxy<TSchema extends MapNodeSchema>(
  * @param customTargetObject - Target object of the proxy.
  * If not provided `{}` is used for the target.
  * @remarks
- * Because this proxy is backed by a raw node, it has the same limitations as the node created by {@link createRawObjectNode}.
- * Most if its properties and methods will error if read/called.
+ * Because this proxy is backed by a raw node, it has the same limitations as the node created by {@link createRawNode}.
+ * Most of its properties and methods will error if read/called.
  */
-export function createRawObjectProxy<TSchema extends ObjectNodeSchema>(
+export function createRawNodeProxy<TSchema extends ObjectNodeSchema>(
+	schema: TSchema,
+	content: InsertableTypedNode<TSchema>,
+	allowAdditionalProperties: boolean,
+	target?: object,
+): Unhydrated<TreeObjectNode<TSchema>>;
+export function createRawNodeProxy<TSchema extends FieldNodeSchema>(
+	schema: TSchema,
+	content: InsertableTypedNode<TSchema>,
+	allowAdditionalProperties: boolean,
+	target?: object,
+): Unhydrated<TreeListNodeOld<TSchema["info"]["allowedTypes"]>>;
+export function createRawNodeProxy<TSchema extends MapNodeSchema>(
 	schema: TSchema,
 	content: InsertableTypedNode<TSchema>,
 	allowAdditionalProperties: boolean,
 	target?: object,
-): Unhydrated<TreeObjectNode<TSchema>> {
+): Unhydrated<TreeMapNode<TSchema>>;
+export function createRawNodeProxy<TSchema extends TreeNodeSchema>(
+	schema: TSchema,
+	content: InsertableTypedNode<TSchema>,
+	allowAdditionalProperties: boolean,
+	target?: object,
+): Unhydrated<TreeNode> {
 	// Shallow copy the content and then add the type name symbol to it.
-	const contentCopy = { ...content };
-	Object.defineProperty(contentCopy, typeNameSymbol, { value: schema.name });
-	const proxy = createObjectProxy(schema, allowAdditionalProperties, target);
-	const editNode = createRawObjectNode(schema, contentCopy);
+	let editNode: RawTreeNode<TSchema, InsertableTypedNode<TreeNodeSchema>>;
+	let proxy: TreeNode;
+	if (schema instanceof ObjectNodeSchema) {
+		const contentCopy = copyContent(schema.name, content as InsertableTypedNode<typeof schema>);
+		editNode = createRawNode(schema, contentCopy);
+		proxy = createObjectProxy(schema, allowAdditionalProperties, target);
+	} else if (schema instanceof FieldNodeSchema) {
+		const contentCopy = copyContent(schema.name, content as InsertableTypedNode<typeof schema>);
+		editNode = createRawNode(schema, contentCopy);
+		proxy = createListProxy(allowAdditionalProperties, target);
```

Comment:
Nit: might be worth a quick comment noting why `FieldNodeSchema` implies lists here

Reply:
Okay